frontend/src/stimulus/controllers/dynamic/storages/oauth-client-form.controller.ts

```diff
@@ -30,27 +30,53 @@
 
 import { Controller } from '@hotwired/stimulus';
 
-export default class StorageFormController extends Controller {
+export default class OAuthClientFormController extends Controller {
   static targets = [
     'clientId',
+    'clientSecret',
     'redirectUri',
+    'submitButton',
   ];
 
   static values = {
     clientIdMissing: String,
   };
 
+  declare readonly hasClientIdTarget:boolean;
+  declare readonly hasClientSecretTarget:boolean;
+  declare readonly hasSubmitButtonTarget:boolean;
   declare readonly clientIdTarget:HTMLInputElement;
-  declare readonly redirectUriTarget:HTMLInputElement;
+  declare readonly clientSecretTarget:HTMLInputElement;
+  declare readonly redirectUriTargets:HTMLInputElement[];
+  declare readonly submitButtonTarget:HTMLInputElement;
 
   declare readonly clientIdMissingValue:string;
 
   connect():void {
-    this.clientIdTarget.addEventListener('input', () => {
-      this.redirectUriTarget.value = this.redirectUri();
-    });
+    this.setRedirectUriValue();
+    this.toggleSubmitButtonDisabled();
+  }
+
+  public toggleSubmitButtonDisabled():void {
+    const targetsConfigured = this.hasClientIdTarget && this.hasClientSecretTarget && this.hasSubmitButtonTarget;
+
+    if (!targetsConfigured) {
+      return;
+    }
 
-    this.redirectUriTarget.value = this.redirectUri();
+    if (this.clientIdTarget.value === '' || this.clientSecretTarget.value === '') {
```

Comment:
It is somehow inconsistent that you disable the button here but nowhere else. On the other places you let the user press the button and render the error message if there is a value missing. I think this would be fine here as well.

Reply:
I spent time on this today and it's not a simple change as we'll have to rework the OAuth Clients service workflow to accommodate this behaviour. Hence, I broker it down in to a [CODE MAINTENANCE wp](https://community.openproject.org/wp/51033)